src/PrestaShopBundle/Form/Admin/Type/EventListener/DisablingToggleListener.php

```diff
@@ -0,0 +1,96 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type\EventListener;
+
+use PrestaShopBundle\Form\Admin\Extension\DisablingToggleExtension;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\FormCloner;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+class DisablingToggleListener implements EventSubscriberInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'addDisablingToggle',
+            FormEvents::PRE_SUBMIT => 'addDisablingToggle',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function addDisablingToggle(FormEvent $event): void
+    {
+        $form = $event->getForm();
+        $parent = $form->getParent();
+        $fieldName = DisablingToggleExtension::FIELD_PREFIX . $form->getName();
+        if ($parent->has($fieldName)) {
+            return;
+        }
+        $parent->add($fieldName, SwitchType::class, [
+            'label' => false,
+            'attr' => [
+                'container_class' => 'disabling-toggle',
+                'data-value-type' => 'boolean',
+            ],
+            'row_attr' => [
+                'class' => 'ps-disabling-toggle',
+                'data-disabling-toggle-name' => $fieldName,
+            ],
+        ]);
+
+        $shouldBeDisabled = !$parent->get($fieldName)->getData();
+
+        $formCloner = new FormCloner();
+        $newOptions = [
+            'disabled' => $shouldBeDisabled,
+            'attr' => [
+                'data-toggled-by' => $fieldName,
+            ],
+        ];
+
+        foreach ($form->all() as $childForm) {
+            $config = $childForm->getConfig();
+            if ($shouldBeDisabled === $config->getOption('disabled', false)) {
+                continue;
+            }
+            $newChildForm = $formCloner->cloneForm($childForm, array_merge($childForm->getConfig()->getOptions(), $newOptions));
+            $form->add($newChildForm);
+        }
+
+        //@todo; need configurable (e.g. it should be possible to change if input is disabled when checkbox is checked or when unchecked
```

Comment:
So if I make it configurable, then it means the option `disabling_toggle` should become an array? What if i want just to add a toggle with default configuration? So could it be bool|array (also the array options couldn't be resolved probably)? example
```
'disabling_toggle' => true // input with default configs
'disabling_toggle' => false // no need for disabling toggle
'disabling_toggle' => [
    'disable_on_match' => true|false,
    'matching_value' => 1
],
```

Reply:
It will be easier to use and understand the feature if `disabling_toggle` remains a simple boolean However you can add another option in the same `DisablingToggleExtension` extension Which could be called `disabling_toggle_options` for example, and this option could use a normaliser and a sub OptionResolver to make sure it respects the expected format (including default values)

Similar to what was done here https://github.com/PrestaShop/PrestaShop/blob/69fcb559c739314e4b2260efced93ad3ba03f36d/src/PrestaShopBundle/Form/Admin/Extension/ExternalLinkExtension.php#L62 Except that the  default`null` value would mean using the default configuration (if `disabling_toggle` i true of course)